DLA-Future Documentation (#8)
RMeli/cp2k: docs/libraries/eigensolvers/dlaf.md

# DLA-Future

> [DLA-Future] is a distributed linear algebra library implemented using the C++ `std::execution` proposal [P2300] as provided by [pika].

Reviewer: This could perhaps also simply refer to "implemented using the C++26 `std::execution` library (link to https://eel.is/c++draft/#exec) as provided by [pika]"? P2300 is, however, good as a reference too, and I'm aware this is still what we refer to in first paragraph on pikacpp.org.

Author: P2300 is also referred in the `pika`-specific paragraph. I apply your suggestion at the top level and re-phrase the `pika`-specific paragraph.

DLA-Future provides a ScaLAPACK-like Fortran interface in [DLA-Future-Fortran](https://github.com/eth-cscs/DLA-Future-Fortran), which can be used as a drop-in replacement for ScaLAPACK (with a subset of ScaLAPACK arguments, e.g. workspace arguments are not present).

## Dependencies

[DLA-Future] has several dependencies. It is officially distributed via the [Spack] package manager, which is the recommended installation option. The [DLA-Future Spack package] lists all the required and optional dependencies. 

## CMake

[DLA-Future] is enabled in CP2K with the following CMake option:

```bash
-DCP2K_USE_DLAF=ON
```

## CP2K Input File

[DLA-Future] can be selected in the CP2K input file with the `PREFERRED_DIAG_LIBRARY` keyword:
```
&GLOBAL
  PREFERRED_DIAG_LIBRARY DLAF
  DLAF_NEIGVEC_MIN 1024
  [...]
&END GLOBAL
```

> The `DLAF_NEIGVEC_MIN` indicates the minimum matrix size for which DLA-Future is used instead of ScaLAPACK. DLA-Future is efficient for large matrices, but it might be slower than ScaLAPACK for small matrices.

Reviewer: Perhaps a bit more clarification in this direction:
```suggestion
The `DLAF_NEIGVEC_MIN` indicates the minimum matrix size for which DLA-Future is used instead of ScaLAPACK. DLA-Future is optimized for large matrices on GPUs, and might be slower than ScaLAPACK for small matrices or CPU matrices.
```
?
